internal/resources/iampolicy/resource_iam_policy.go

```diff
@@ -0,0 +1,423 @@
+				ForceNew:    true,
+			},
+			managementClusterNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of ManagementCluster",
+				Default:     "attached",
+				Optional:    true,
+				ForceNew:    true,
+			},
+			provisionerNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of Provisioner",
+				Default:     "attached",
+				Optional:    true,
+				ForceNew:    true,
+			},
+			clusterNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of Cluster",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	},
+}
+
+var roleBinding = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "List of role bindings associated with the policy",
+	Required:    true,
+	ForceNew:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			roleKey: {
+				Type:        schema.TypeString,
+				Description: "Role for this rolebinding -max length for role is 126",
+				Required:    true,
+				ForceNew:    true,
+			},
+			subjectsKey: {
+				Type:        schema.TypeList,
+				Description: "Subject of rolebinding",
+				Required:    true,
+				ForceNew:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						subjectNameKey: {
+							Type:        schema.TypeString,
+							Description: "Subject name - allow max characters for email - 320",
+							Required:    true,
+						},
+						subjectKindKey: {
+							Type:        schema.TypeString,
+							Description: "Subject type",
+							Required:    true,
+						},
+					},
+				},
+			},
+		},
+	},
+}
+
+func constructRoleBindingList(d *schema.ResourceData) []*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding {
+	var rbl = make([]*iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding, 0)
+
+	value, ok := d.GetOk(roleBindingKey)
+	if !ok {
+		return rbl
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return rbl
+	}
+
+	for _, raw := range data {
+		rbData := raw.(map[string]interface{})
+
+		var rb = &iammodel.VmwareTanzuCoreV1alpha1PolicyRoleBinding{
+			Subjects: make([]*iammodel.VmwareTanzuCoreV1alpha1PolicySubject, 0),
+		}
+
+		if v, ok := rbData[roleKey]; ok {
+			helper.SetPrimitiveValue(v, &rb.Role, roleKey)
+		}
+
+		if v, ok := rbData[subjectsKey]; ok {
+			subjects, _ := v.([]interface{})
+			for _, sb := range subjects {
+				rb.Subjects = append(rb.Subjects, expandSubjects(sb))
+			}
+		}
+
+		rbl = append(rbl, rb)
+	}
+
+	return rbl
+}
+
+func constructBindingDeltaList(d *schema.ResourceData, op *iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDeltaOpType) []*iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDelta {
+	rbList := constructRoleBindingList(d)
+
+	var deltaList []*iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDelta
+
+	for _, rb := range rbList {
+		for _, sub := range rb.Subjects {
+			deltaList = append(
+				deltaList,
+				&iammodel.VmwareTanzuCoreV1alpha1PolicyBindingDelta{
+					Role:    rb.Role,
+					Subject: sub,
+					Op:      op,
```

Comment:
For create and delete the OP will remain same for the whole delta list but this would not be the case for update, as the update delta list will have a mix of add and remove op.

Reply:
Yes, handled in the update accordingly.